clippy_lints/src/unit_types/let_unit_value.rs

```diff
@@ -71,25 +71,38 @@ pub(super) fn check<'tcx>(cx: &LateContext<'tcx>, local: &'tcx LetStmt<'_>) {
                 local.span,
                 "this let-binding has unit value",
                 |diag| {
+                    let mut suggestions = Vec::new();
+
+                    // Suggest omitting the `let` binding
                     if let Some(expr) = &local.init {
                         let mut app = Applicability::MachineApplicable;
                         let snip = snippet_with_context(cx, expr.span, local.span.ctxt(), "()", &mut app).0;
-                        diag.span_suggestion(local.span, "omit the `let` binding", format!("{snip};"), app);
+                        suggestions.push((local.span, format!("{snip};")));
                     }
 
-                    if let PatKind::Binding(_, binding_hir_id, ident, ..) = local.pat.kind
+                    // If this is a binding pattern, we need to add suggestions to remove any usages
+                    // of the variable
+                    if let PatKind::Binding(_, binding_hir_id, ..) = local.pat.kind
                         && let Some(body_id) = cx.enclosing_body.as_ref()
-                        && let body = cx.tcx.hir().body(*body_id)
-                        && is_local_used(cx, body, binding_hir_id)
```

Comment:
This line is completely removed, is this intended?

Reply:
And sorry for the late review, I've opened this PR maybe 5 times and never got around to clicking that "Submit review" button

Reply:
Hey @blyxyas no worries!

I think it is unnecessary now; this part here collects usages:

https://github.com/rust-lang/rust-clippy/blob/ec243887d3524545922e340a4e55c4bd174e966e/clippy_lints/src/unit_types/let_unit_value.rs#L90-L92

... and we then emit a lint based on whether or not we had any: 

https://github.com/rust-lang/rust-clippy/blob/ec243887d3524545922e340a4e55c4bd174e966e/clippy_lints/src/unit_types/let_unit_value.rs#L99-L105

It could be added back as a sort of "performance optimization guard" to remove the need to call `walk_body`, assuming that `is_local_used` is quicker, but it doesn't change the behaviour (I also double checked this!)

Reply:
I've been all mornig trying to measure this but it's in a very awkward spot to get a hold of how many heavy-lifting that function does.

Being that in the case that the body is visited is already in the linting-case, let's just ignore it.